pkg/handlers/scan.go

```diff
@@ -0,0 +1,54 @@
+package handlers
+
+import (
+	"encoding/json"
+	"errors"
+	cmmn "github.com/kptm-tools/common/common/events"
+	"github.com/kptm-tools/core-service/pkg/api"
+	"github.com/kptm-tools/core-service/pkg/interfaces"
+	"net/http"
+)
+
+type ScanHandlers struct {
+	scanService interfaces.IScanService
+	eventBus    cmmn.EventBus
+}
+
+var _ interfaces.IScanHandlers = (*ScanHandlers)(nil)
+
+func NewScanHandlers(scanService interfaces.IScanService, bus cmmn.EventBus) *ScanHandlers {
+	return &ScanHandlers{
+		scanService: scanService,
+		eventBus:    bus,
+	}
+}
+
+func (s ScanHandlers) CreateScans(w http.ResponseWriter, req *http.Request) error {
+
+	scanRequest := new(ScanRequest)
+
+	if err := decodeJSONBody(w, req, scanRequest); err != nil {
+		var mr *malformedRequest
+
+		if errors.As(err, &mr) {
+			return api.WriteJSON(w, mr.status, api.APIError{Error: mr.Error()})
+		} else {
+			return api.WriteJSON(w, http.StatusInternalServerError, api.APIError{Error: err.Error()})
+		}
+	}
+
+	scan, err := s.scanService.CreateScans(scanRequest.HostIds)
+	scanStartedPayload := &cmmn.ScanStartedEvent{
+		ScanID:    scan.ID,
+		Targets:   scan.Targets,
+		Timestamp: scan.CreatedAt.Unix(),
+	}
+	scanStartedString, err := json.Marshal(scanStartedPayload)
```

Comment:
My bad here for giving you poor example code. Maybe I'm being a bit pedantic, but scanStartedString is of type []byte. So the correct naming of this variable should be:
```suggestion
	scanStartedBytes, err := json.Marshal(scanStartedPayload)
```

Reply:
Corrected